relay-server/src/actors/spooler/mod.rs

```diff
@@ -252,6 +264,24 @@ impl InMemory {
         self.buffer.values().map(|v| v.len()).sum()
     }
 
+    /// Runs the check for in-memory buffer and emits the metrics and/or errors.
+    fn check(&self) {
+        let count = self.buffer.keys().len();
+        relay_statsd::metric!(gauge(RelayGauges::BufferProjectsMemoryCount) = count as u64);
+
+        // Go over the buffered emvelopes and check if any of the keys exceeds the threshold.
+        for (key, values) in &self.buffer {
+            let number_of_envelopes = values.len();
+            if number_of_envelopes > NUMBER_OF_ENVELOPES_PER_KEY {
+                relay_log::error!(
+                    key = %key,
```

Comment:
So we can group by key in sentry UI:

```suggestion
                    tags.key = %key,
```